ee/be/modules/350-node-local-dns/images/stale-sockets-cleaner/main.go

```diff
@@ -0,0 +1,288 @@
+/*
+Copyright 2024 Flant JSC
+Licensed under the Deckhouse Platform Enterprise Edition (EE) license. See https://github.com/deckhouse/deckhouse/blob/main/ee/LICENSE
+*/
+
+package main
+
+import (
+	"context"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"net"
+	"os"
+	"syscall"
+	"time"
+
+	"github.com/vishvananda/netlink/nl"
+	"golang.org/x/sys/unix"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/vishvananda/netlink"
+)
+
+const (
+	nldNS               string = "kube-system"
+	nldLabelSelector    string = "app=node-local-dns"
+	nldDstPort          uint16 = 53
+	scanInterval               = 30 * time.Second
+	sizeofSocketID             = 0x30
+	sizeofSocketRequest        = sizeofSocketID + 0x8
+	SOCK_DESTROY               = 21
+	familyIPv4                 = syscall.AF_INET
+	protoUDP                   = unix.IPPROTO_UDP
+)
+
+var (
+	native       = nl.NativeEndian()
+	networkOrder = binary.BigEndian
+)
+
+/*
+When cni-cilium and node-local-dns are enabled, and pod node-local-dns has been restarted, we may encounter problems with dns.
+This is because the udp socket remains active in the application pods until the ip address of the pod, which has already been deleted.
+To prevent this problem, we perform the following actions:
+- We get the name and PodCidr of the node on which the application is running
+- Then every 30 seconds
+  - We get the current IP address of the node-local-dns pod
+  - We get all UDP sockets on the node, and look among them for those with dst_port 53, and dsp_ip belongs to PodCidr, but it is not equal to the current ip address of the node-local-dns pod
+  - If we find such sockets, then we delete them.
```

Comment:
```suggestion
It is the workaround for issue https://github.com/cilium/cilium/issues/31012.
When cni-cilium and node-local-dns modules are enabled, and node-local-dns pod has been restarted, we may encounter problems with staled dns connections.
This is because the udp socket remains active in the application pods with the dst ip address of the old node-local-dns pod, which has already been deleted.
To prevent this problem, we perform the following actions:
- We get the name and PodCidr of the node on which the application is running.
- Then every 30 seconds:
  - We get the current IP address of the node-local-dns pod.
  - We get all UDP sockets on the node, and look among them for those with dst_port 53, and dsp_ip belongs to PodCidr, but it is not equal to the current ip address of the node-local-dns pod.
  - If we find such sockets, then we delete them.
```